### PR TITLE
Add verify-error state and template to amp-form

### LINKED ISSUE #16728: verify-error for forms
## What's the issue?

We would like to render a div when xhr form validation fails, similar to ```<div submit-error>```

## How do we reproduce the issue?
```
<div verify-error>
  <template type="amp-mustache">
    {{"{{#verifyErrors}}"}}
      <p>{{message}}</p>
    {{"{{/verifyErrors}}"}}
    {{"{{^verifyErrors}}"}}
      <p>Something went wrong. Try again later?</p>
    {{"{{/verifyErrors}}"}}
  </template>
</div>
```

## What browsers are affected?

All.

## Which AMP version is affected?

1531357871900

### PR DESCRIPTION
**Pull request by cvializ**
Fixes #16728

Follow the same API as the data structure in `submit-error`.

/to @choumx @aghassemi @cathyxz
(can't add as reviewers rn from the slow plane wifi)

### CODE REVIEW

File: extensions/amp-form/0.1/amp-form.js [hunk 1]
```diff
@@ -268,21 +269,18 @@ export class AmpForm {
...
-      }
-    };
     this.form_.addEventListener('change', e => {
       this.verifier_.onCommit()
-          .then(updatedElements => {
+          .then(({updatedElements, errors}) => {
+            if (this.state_ === FormState_.VERIFYING) {
+              this.setState_(errors.length ?
+                FormState_.VERIFY_ERROR : FormState_.INITIAL);
+              this.renderTemplate_(
```
Comment (aghassemi) [inline extensions/amp-form/0.1/amp-form.js]: I suspect template should not be rendered if `errors.length == 0`?

**Comment by choumx** [comment]
Also need changes to validator-main.protoascii with tests.

File: extensions/amp-form/0.1/amp-form.js [hunk 2]
```diff
@@ -268,21 +267,24 @@ export class AmpForm {
...
+      this.verifier_.onCommit().then(({updatedElements, errors}) => {
+        // Only make the verify XHR if the user hasn't pressed submit.
+        if (this.state_ === FormState.VERIFYING) {
+          if (errors.length) {
+            this.setState_(FormState.VERIFY_ERROR);
+            this.renderTemplate_(
+                /** @type {!JsonObject} */ ({verifyErrors: errors}));
+            this.triggerAction_(FormEvent.VERIFY_ERROR, errors);
+          } else {
+            this.setState_(FormState.INITIAL);
```
Comment (choumx) [inline extensions/amp-form/0.1/amp-form.js]: Any consequence of changing this order of operations?

**Comment by aghassemi** [comment]
Also in amp-mustache. Can be part of https://github.com/ampproject/amphtml/issues/16215. otherwise LGTM

File: extensions/amp-form/0.1/amp-form.js (hunk 2 above)
Comment (cvializ) [reply extensions/amp-form/0.1/amp-form.js]: Oops you're right the state should be changed after all the elements have been updated.

**Comment by cvializ** [comment]
@aghassemi @choumx PTAL

**Pull request merged**